R/dotproduct.R

```diff
@@ -0,0 +1,73 @@
+#' @name dotproduct
+#' @title Calculate the dot product or normalized dot product
```

Comment:
Only a very minor comment: I would prefer to have a newline before every `@...` roxygen entry - just for easier readability

Reply:
Changed.